### PR TITLE
docs: update readme

### PR DESCRIPTION
**Pull request by weichweich**
## related https://github.com/KILTprotocol/ticket/issues/2126

Update the Readme. The content of the readme should moved to docs.kilt.io but since this PR was already open i suggest that we merge that and move the docs later.

I removed the react and vue examples as they are not mentioned it the readme and are not tested, but we should include them when we move this to docs.kilt.io (A ticket for this is already created: https://github.com/KILTprotocol/ticket/issues/2822)

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1,89 @@
...
+    <>
+      <h2>Extensions</h2>
+      <ul>
+        {extensions.map((ext, i) => (
+          <li key={i}>{ext.name}</li>
+        ))}
+      </ul>
+    </>
+  )
+}
```
Comment (Dudleyneedham) [inline README.md:59]: Do you want to add TS instead, or maybe remove all the types?
```suggestion
```ts
import { watchExtensions, Types } from 'kilt-extension-api'

export default function Home(): JSX.Element {
  const [extensions, setExtensions] = useState<
    Types.InjectedWindowProvider<Types.PubSubSessionV1 | Types.PubSubSessionV2>[]
  >([])

  useEffect(() => {
    watchExtensions((extensions) => {
      setExtensions(extensions)
    })
  }, [])

  return (
    <>
      <h2>Extensions</h2>
      <ul>
        {extensions.map((ext, i) => (
          <li key={i}>{ext.name}</li>
        ))}
      </ul>
    </>
  )
}
```

**Review by Dudleyneedham — CHANGES_REQUESTED** [review]
Minor thing

File: README.md (hunk 1 above)
Comment (weichweich) [reply README.md]: [ed51ec2](https://github.com/KILTprotocol/kilt-extension-api/pull/15/commits/ed51ec28238b81c2ee247d629ca98ea1f6fb9e2d)

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Pull request merged**